projects/core/src/checkout/http-interceptors/checkout-cart.interceptor.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  HttpErrorResponse,
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, throwError } from 'rxjs';
+import { catchError, switchMap, take } from 'rxjs/operators';
+import { MultiCartService } from '../../cart/facade/multi-cart.service';
+import { RouterState, RoutingService } from '../../routing/index';
+
+@Injectable({ providedIn: 'root' })
+export class CheckoutCartInterceptor implements HttpInterceptor {
+  constructor(
+    protected routingService: RoutingService,
+    protected multiCartService: MultiCartService
+  ) {}
+
+  intercept(
+    request: HttpRequest<any>,
+    next: HttpHandler
+  ): Observable<HttpEvent<any>> {
+    return this.routingService.getRouterState().pipe(
+      take(1),
+      switchMap((state: RouterState) => {
+        return next.handle(request).pipe(
+          catchError((response: any) => {
+            if (
+              response instanceof HttpErrorResponse &&
+              this.isCheckoutUserInCheckout(state.state?.semanticRoute)
+            ) {
+              if (
+                response.status === 400 &&
```

Comment:
Either this check can be part of the isCartNotFoundError function or you can get rid of the function and just put the contents directly here (for more clarity)